tests/checker/good/lib_typing2.scilla

```diff
@@ -0,0 +1,28 @@
+scilla_version 0
+
+(* The purpose of this test contract is to demonstrate that *)
+(* type errors in variable declarations in libraries are *)
+(* handled by the type checker. *)
+
+library BadLibType
```

Comment:
Can we change this name, please? It's a positive test case, so it would be preferable if the library name didn't suggest otherwise. :-)